src/internal_events/common.rs

```diff
@@ -127,3 +124,26 @@ impl InternalEvent for CollectionCompleted {
         histogram!("collect_duration_seconds", self.end - self.start);
     }
 }
+
+#[derive(Debug)]
+pub struct ComponentEventsDropped {
+    pub count: u64,
+    pub intentional: bool,
+    pub reason: &'static str,
+}
+
+impl InternalEvent for ComponentEventsDropped {
+    fn emit(self) {
+        let intent = if self.intentional { "true" } else { "false" };
+        error!(
+            message = "Events dropped.",
+            intentional = intent,
```

Comment:
I think in another PR we've started preferring boolean values in the `error!()` call. Metric tags require string values though so you'll still need the mapping for the `counter!()`.